### DISCUSSION TITLE
TikTok lagging when opened, no issues with any other Apps

### DISCUSSION THREAD

**Discussion by holgeight** [appium/appium#19255] [Q&A]
Hey, so I'm attempting to run some very simple automation on my iphone. I've noted for some reason, when I open TikTok through Appium, the App freezes for about 3-4 minutes before it's useable. I don't have this issue with any other App. I'm just curious if anybody else has ran into this issue with TikTok or any other App and found a fix? 

It seems to work fine after that initial 3-5 minute freeze.

**Comment by nathandev0** [comment]
Hey, have you been able to resolve the issue?

**Comment by ignat-feofanov** [comment]
привет, получилось что-то решить?

**Comment by somethingintheway1** [comment]
решил?

**Comment by Heyh520** [comment]
I also have this problem, I have tried many methods but to no avail, have you found a solution?

**Reply by GGBond6** [reply]
me too

**Reply by YoussefEssalhi** [reply]
@Heyh520 @GGBond6 is there any news

**Comment by mykola-mokhnach** [comment]
Might be related to https://github.com/appium/appium-xcuitest-driver/blob/master/docs/guides/wda-slowness.md#pattern-application-startup-is-slow

**Comment by shatk0vskiy** [comment]
Any news?

**Comment by YoussefEssalhi** [comment]
Up

**Comment by eglitise** [comment]
I believe TikTok loads a massive amount of elements in the initial page source. You could mitigate this using the `snapshotMaxDepth` setting, by setting it to a low value (10-15) [when starting the session:](https://appium.io/docs/en/latest/guides/settings/#initializing-settings-via-capabilities)
```
{
    "appium:settings[snapshotMaxDepth]": 10
}
```
Then you can navigate to your desired page and reset `snapshotMaxDepth` back to the default 50, or any other value.
